lib/grpc/client/adapters/gun.ex

```diff
@@ -135,7 +137,37 @@ defmodule GRPC.Client.Adapters.Gun do
   end
 
   @impl true
-  def recv_headers(%{conn_pid: conn_pid}, %{stream_ref: stream_ref}, opts) do
+  def receive_data(
+        %{server_stream: true, channel: %{adapter_payload: adapter_payload}, payload: payload} =
+          stream,
+        opts
+      ) do
+    with {:ok, headers, is_fin} <- recv_headers(adapter_payload, payload, opts),
+         response <- response_stream(is_fin, stream, opts) do
```

Comment:
since you don't match on anything specific for `response`, I think it's better to move it inside the `do` block